msg.go

```diff
@@ -388,6 +401,9 @@ Loop:
 			if ptr++; ptr > 10 {
 				return "", lenmsg, &Error{err: "too many compression pointers"}
 			}
+			// pointer should guarantee that it advances and points forwards at least
+			// but the condition on previous three lines guarantees that it's
+			// at least loop-free
 			off = (c^0xC0)<<8 | int(c1)
```

Comment:
BIND and Knot don't seem to accept forward pointers (if I've read their code correctly), maybe  that implementation has died out?